app/updatemanager.go

```diff
@@ -149,9 +150,18 @@ type UpdateControlMapState struct {
 }
 
 type UpdateControlMap struct {
-	ID       string                           `json:"id"`
-	Priority int                              `json:"priority"`
-	States   map[string]UpdateControlMapState `json:"states"`
+	ID               string                           `json:"id"`
+	Priority         int                              `json:"priority"`
+	States           map[string]UpdateControlMapState `json:"states"`
+	lastModifiedTime time.Time
+	expiryTime       time.Time
+}
+
+func (u *UpdateControlMap) Stamp(updateControlTimeoutSeconds int) *UpdateControlMap {
+	u.lastModifiedTime = time.Now()
+	u.expiryTime = u.lastModifiedTime.Add(
+		time.Duration(updateControlTimeoutSeconds) * time.Second)
```

Comment:
If both `lastModifiedTime` and `expiryTime` are set only here, we should be able to remove one of them. :thinking:

Reply:
Yes, I've been thinking the same, depends a bit on what is actually needed down the line